bitnami/fluentd/templates/_helpers.tpl

```diff
@@ -39,6 +39,8 @@ app.kubernetes.io/name: {{ include "fluentd.name" . }}
 helm.sh/chart: {{ include "fluentd.chart" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/version: {{ .Chart.AppVersion }}
+version: {{ .Chart.AppVersion }}
```

Comment:
Please undo these changes. Labels including versions are not compatible with upgrades (we found several issues in the past)

More context: 

This label will be different depending on the chart version. So, when you upgrade from version `X.X.X` to `Y.Y.Y`, the k8s don't use the same labels. This is highly problematic when dealing with PVCs, replicasets, etc.

Reply:
Istio doesn't need the `version` label to be exactly `{{ .Chart.AppVersion }}` at all but it does need to have a `version` label to avoid emitting validation warnings in Kiali user interface because it is missing.
I added a `version: v1` label as default which is a reasonable compromise to me.

Reply:
Please, you still need to remove `app.kubernetes.io/version: {{ .Chart.AppVersion }}`.

if you need to take the risk of having version labels we could do implementing another approach like:

https://github.com/bitnami/charts/blob/a0f9f97ed0a05ad6e26ee6ad492326830afb9f67/bitnami/mongodb-sharded/values.yaml#L597

https://github.com/bitnami/charts/blob/a0f9f97ed0a05ad6e26ee6ad492326830afb9f67/bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml#L18-L22

you could use a `extraLabels` or even `aggregator.extraLabels`

Reply:
I don't see any risk in having a `version: v1` default label applied on Pods which would be constant and never updated unless canary deployments occurs.
Sorry I missed the `app.kubernetes.io/version: {{ .Chart.AppVersion }}` and I'm going to remove it

Reply:
I wouldn't like to add `version: v1` to the chart if it means nothing, why v1? a version for the chart? the deployment? the app? it is a bit confusing and I agree with @juan131 it could create conflict. 

Having the second option is a good point so that you could add this label if you need and then you know why and your reasons and we keep the chart decoupled from this Kiali user interface.

Reply:
I need to understand exactly what kind of conflicts we're talking about.
I totally agree that second option would allow having extra needed labels regardless from the `version`, but if there is another option. 
Say for example I add `version: {{ .Chart.AppVersion }}` label directly to the Pod template as already done for the [app label](https://github.com/irizzant/charts-1/blob/fdd5b293001af3f8fd355346960a2702db3d9d57/bitnami/fluentd/templates/aggregator-statefulset.yaml#L9).
I deploy the chart with version X.X.X and:
* PVC are created and storage is claimed using only `Values.aggregator.persistence.accessMode`, `Values.aggregator.persistence.storageClass` and `.Values.aggregator.persistence.size` values
* StatefulSet, ReplicaSet, DaemonSet all use `spec.selector.matchLabels` to select the Pod they act on

After a while I switch the chart to Y.Y.Y version and:
* PVC keep on being tied to the right PV as per StatefulSet contract
* StatefulSet, ReplicaSet, DaemonSet will recreate their Pod according to their  `spec.selector.matchLabels`.

Given the above, I don't see any conflict rising.

Reply:
In addition to the above, I've just pushed a commit for the necessary changes.
At this point I'd just add the `version: {{ .Chart.AppVersion }}` label to the `template.metadata.labels`.

Reply:
Cool! thanks!